script/vsts/nightly-release.yml

```
@@ -51,7 +51,7 @@ jobs:
         displayName: Download Release Artifacts
 
       - script: |
-          node $(Build.SourcesDirectory)\script\vsts\upload-artifacts.js --create-github-release --assets-path "$(System.ArtifactsDirectory)" --linux-repo-name "atom"
+          node $(Build.SourcesDirectory)\script\vsts\upload-artifacts.js --create-github-release --assets-path "$(System.ArtifactsDirectory)" --skip-upload-to-s3 $(SKIP_UPLOAD_TO_S3)
```

Why does this remove 
```suggestion
          node $(Build.SourcesDirectory)\script\vsts\upload-artifacts.js --create-github-release --assets-path "$(System.ArtifactsDirectory)" --linux-repo-name "atom" --skip-upload-to-s3 $(SKIP_UPLOAD_TO_S3)
```
We should keep the code the same the upstream and us. If this flag does not break the code, we should not remove it.

---

So, there are two ways forward with that flag, assuming we want to have the same code as upstream.

1. Add another flag `--skip-upload-to-packagecloud $(SKIP_UPLOAD_TO_PACKAGECLOUD)` (or whatever name we pick for the flag/variable)
2. Something I mentioned in a comment above. Don't add these new flags, and simply look for `process.env.SKIP_UPLOAD_TO_S3` and `process.env.SKIP_UPLOAD_TO_PACKAGECLOUD` from javascript in `upload-artifacts.js`

I personally am hoping to do the second approach of those two, but I wanted to get your opinion before rewriting the PR so dramatically, compared to the version you already saw and commented on.

---

Why not use a single `process.env.SKIP_UPLOAD_TO_PACKAGECLOUD` to skip both?

---

Sure, I suppose. We at this fork don't need it to be more granular than that.

If I understand correctly, the S3 bucket is what backs downloads from atom.io, like this: https://atom.io/download/deb

And the PackageCloud.io one is useful because you can add it as an apt/rpm repo. 

```bash
curl -s https://packagecloud.io/install/repositories/AtomEditor/atom/script.deb.sh | sudo bash
```

or direct download: https://packagecloud.io/AtomEditor/atom/packages/any/any/atom_1.49.0_amd64.deb

In theory, one could want this split up so they could enable just PackageCloud. Or just enable their own S3 bucket.

Somehow that feels more correct to me, to have it be two variables for two separate services.

We could do all three.

`process.env.SKIP_UPLOAD === 'true'` could enable both,

`process.env.SKIP_UPLOAD_TO_PACKAGECLOUD !== 'true'` could individually override that and enable just PackageCloud...

As always, lots of ways to handle booleans, we just should pick one.

(If you have no opinion, I'm still leaning toward two env variables.)

---

If you strongly prefer it to be one variable, I will do that.

I think it makes sense for two variables, since either service (PackageCloud.io account vs an S3 bucket) could be set up independent of the other. 

But the important thing is we can toggle them off.

(Long-term plans: Ultimately if upstream has a preference I would be inclined to do it the way they like. And we can adopt that formatting of it in this fork.)

---

I prefer one variable because it makes the code cleaner for upstream. I think they prefer fewer variables in their CI too. That just simplifies things. If the code was all ours, I would have leaned towards the more flexibility approach.

---

Thinking forward (as we are already doing here) to upstream hypothetically reviewing and merging this:

If upstream has a different preference for how to do this, would you be comfortable generally speaking migrating to that? I personally am willing to go with whatever upstream wants (if it meets our basic needs).

I ask, because that means the stakes are so low as to be almost nothing and I'll just merge something we can use. With stakes that low, I can use your suggestion verbatim, no problem at all, then ask upstream what they think.

The stakes for us, long-term, would be if we ever wanted to use either of those services. In which case, we might have missed the opportunity to make this as flexible as we need it without diverging from upstream. I admit that's a hypothetical, so right now I will still go with the "one variable" -- so long as we can switch to what upstream wants if they want something different.

Something to keep in mind: If we can convince upstream default these services to "off", then our variable use becomes simpler (just don't set them), but makes theirs more complicated, so I'm not even sure if it's polite to ask them to do that.

---

Maybe we should document every CI variable in `script/vsts/README.md`? I think we should.

---

That single variable does not need to be boolean. It can be a string. For example, 
```
SKIP_UPLOAD="S3, PackageCloud"
```
In JavaScript, this can be broken into two separate strings:
```
"S3, PackageCloud".split(",") // gives an array ["S3", "PackageCloud"].
```

Instead of skipping the upload, we can go the other way around and name the variable `UPLOAD_TO` which gives a list of the services the code should upload artifacts to.

> Maybe we should document every CI variable in script/vsts/README.md? I think we should.

Yes, we should after we fixed this pull request.

---

I acknowledge that that is technically possible, but the "brain being fully turned on" level needed to construct a properly-formed string representing an array of parameters is higher than setting booleans.

I want our CI to be foolproof, because I am a fool.

My 2 cents.

---

That said `indexOf [argument we're looking for]" !== 0` isn't a lot more complex than `var === 'true'`. A bit, but not a lot.